tests/scripts/thread-cert/otci/otci/otci.py

```diff
@@ -0,0 +1,1865 @@
+
+    def get_router_table(self) -> Dict[RouterId, Dict[str, Any]]:
+        """table of routers."""
+        output = self.execute_command('router table')
+        if len(output) < 2:
+            raise UnexpectedCommandOutput(output)
+
+        #
+        # Example output:
+        #
+        # | ID | RLOC16 | Next Hop | Path Cost | LQ In | LQ Out | Age | Extended MAC     |
+        # +----+--------+----------+-----------+-------+--------+-----+------------------+
+        # | 21 | 0x5400 |       21 |         0 |     3 |      3 |   5 | d28d7f875888fccb |
+        # | 56 | 0xe000 |       56 |         0 |     0 |      0 | 182 | f2d92a82c8d8fe43 |
+        # Done
+        #
+
+        table = {}
+        for line in output[2:]:
+            line = line.strip()
+            if not line:
+                continue
+
+            fields = line.split('|')
+            _, id, rloc16, next_hop, path_cost, lq_in, lq_out, age, extaddr, _ = fields
+
+            table[RouterId(id)] = {
+                'id': RouterId(id),
+                'rloc16': Rloc16(rloc16, 16),
+                'next_hop': int(next_hop),
+                'path_cost': int(path_cost),
+                'lq_in': int(lq_in),
+                'lq_out': int(lq_out),
+                'age': int(age),
+                'extaddr': extaddr,
+            }
+
+        return table
+
+    # TODO: router <id>
+
+    #
+    # Router utilities: Child management
+    #
+
+    def get_child_table(self) -> Dict[ChildId, Dict[str, Any]]:
+        """Get the table of attached children."""
+        output = self.execute_command('child table')
+        if len(output) < 2:
+            raise UnexpectedCommandOutput(output)
+
+        #
+        # Example output:
+        # | ID  | RLOC16 | Timeout    | Age        | LQ In | C_VN |R|D|N|Ver|CSL|QMsgCnt| Extended MAC     |
+        # +-----+--------+------------+------------+-------+------+-+-+-+---+---+-------+------------------+
+        # |   1 | 0xc801 |        240 |         24 |     3 |  131 |1|0|0|  3| 0 |     0 | 4ecede68435358ac |
+        # |   2 | 0xc802 |        240 |          2 |     3 |  131 |0|0|0|  3| 1 |     0 | a672a601d2ce37d8 |
+        # Done
+        #
+
+        table = {}
+        for line in output[2:]:
+            line = line.strip()
+            if not line:
+                continue
+
+            fields = line.split('|')
+            _, id, rloc16, timeout, age, lq_in, c_vn, r, d, n, ver, csl, qmsgcnt, extaddr, _ = fields
+
+            r, d, n = int(r), int(d), int(n)
+            mode = DeviceMode(f'{"r" if r else ""}{"d" if d else ""}{"n" if n else ""}')
+            ver, csl, qmsgcnt = int(ver), int(csl), int(qmsgcnt)
+
+            table[ChildId(id)] = {
+                'id': ChildId(id),
+                'rloc16': Rloc16(rloc16, 16),
+                'timeout': int(timeout),
+                'age': int(age),
+                'lq_in': int(lq_in),
+                'c_vn': int(c_vn),
+                'mode': mode,
```

Comment:
intentionally empty string for sleepy device?

Reply:
No, it's a bug